applications/MeshMovingApplication/python_scripts/trilinos_mesh_solver_base.py

```diff
@@ -34,11 +35,20 @@ def __init__(self, mesh_model_part, custom_settings):
     def AddVariables(self):
         super(TrilinosMeshSolverBase, self).AddVariables()
         self.mesh_model_part.AddNodalSolutionStepVariable(KratosMultiphysics.PARTITION_INDEX)
-        self.print_on_rank_zero("::[MeshSolverBase]:: Variables ADDED.")
-
-    def AddDofs(self):
-        super(TrilinosMeshSolverBase, self).AddDofs()
-        self.print_on_rank_zero("::[MeshSolverBase]:: DOFs ADDED.")
+        self.print_on_rank_zero("::[TrilinosMeshSolverBase]:: Variables ADDED.")
+
+    def ImportModelPart(self):
+        self.print_on_rank_zero("::[TrilinosMeshSolverBase]:: ", "Importing model part.")
+        from trilinos_import_model_part_utility import TrilinosImportModelPartUtility
+        self.trilinos_model_part_importer = TrilinosImportModelPartUtility(self.mesh_model_part, self.settings)
+        self.trilinos_model_part_importer.ImportModelPart()
+        self.print_on_rank_zero("::[TrilinosMeshSolverBase]:: ", "Finished importing model part.")
+
+    def PrepareModelPart(self):
```

Comment:
I don't know what the reason is for having `PrepareModelPart` as separate from `ImportModelPart`, but it seems to make the class more delicate to use.

Reply:
short version: If the solver does NOT import the ModelPart (e.g. the holding class, e.g. the optimizer does it), then it needs `PrepareModelPart` to be callable separately
I think the names represent that quite clearly
=> this is also in the baseclass and the wiki

long version: please ask Daniel :)